clients/drcachesim/drpt2trace/pt2ir.cpp

```diff
@@ -300,11 +315,23 @@ pt2ir_t::convert(OUT instrlist_t **ilist)
             decode(GLOBAL_DCONTEXT, insn.raw, instr);
             instr_set_translation(instr, (app_pc)insn.ip);
             instr_allocate_raw_bits(GLOBAL_DCONTEXT, instr, insn.size);
+            instr_set_isa_mode(instr,
+                               insn.mode == ptem_32bit ? DR_ISA_IA32 : DR_ISA_AMD64);
+#ifdef DEBUG
+            /* XXX: Currently, the PT raw data may contain 'STAC' and 'CLAC'
+             * instructions that are not supported by Dynamorio. We'd better to
+             * support them in the future.
+             */
             if (!instr_valid(instr)) {
-                ERRMSG("Failed to convert the libipt's IR to Dynamorio's IR.\n");
-                instrlist_clear_and_destroy(GLOBAL_DCONTEXT, *ilist);
-                return PT2IR_CONV_ERROR_DR_IR_CONVERT;
+                /* We print the invalid instruction‘s PC and raw bytes in DEBUG mode.
+                 */
+                dr_fprintf(STDOUT, "0x%" PRIx64 " <INVALID> <raw ", insn.ip);
```

Comment:
@derekbruening do we want to add this to instr library maybe?

Reply:
@dolanzhao Can you check if we can use instr_disassemble instead for this case? It may already print the raw bytes, so we probably don't need custom logic for printing just the raw bytes.

Reply:
Maybe we can modify instr_disassemble to print the raw bytes in case of invalid instructions (and then use instr_disassemble here). Seems a useful feature to have in general.

Reply:
@derekbruening @abhinav92003 I modified the internal_instr_disassemble() to support printing raw bits for invalid instructions. Could you help me review them?